core/templates/dev/head/domain/exploration/ExplorationObjectFactory.js

```diff
@@ -23,6 +23,8 @@ oppia.factory('ExplorationObjectFactory', [
   function(
       INTERACTION_SPECS, INTERACTION_DISPLAY_MODE_INLINE, StateObjectFactory,
       UrlInterpolationService) {
+    var currentExploration = null;
```

Comment:
Agreed. @maitbayev -- maybe you could create a service to store the state and handle the logic-related stuff? Here is a link to an example of how we've done that: https://github.com/oppia/oppia/blob/simple-editor/core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerService.js